router/pkg/config/config.go

```diff
@@ -685,16 +689,14 @@ func LoadConfig(configFilePath string, envOverride string) (*LoadResult, error)
 
 		configFileBytes = []byte(configYamlData)
 
-		err = ValidateConfig(configFileBytes, JSONSchema)
-		if err != nil {
-			return nil, fmt.Errorf("router config validation error: %w", err)
-		}
-
-		// Unmarshal the final config
-
 		if err := yaml.Unmarshal(configFileBytes, &cfg.Config); err != nil {
 			return nil, err
 		}
+
+		err = cfg.Config.Validate()
```

Comment:
Do I misunderstand this or are we unmarshalling the config twice? once here and another time in validate?

Reply:
It was previously being unmarshalled twice — this PR cuts it down to once.